components/ws-manager-mk2/controllers/subscriber_controller.go

```diff
@@ -80,23 +83,47 @@ func (r *SubscriberReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		}
 	}()
 
+	filterByManagedBy := func(ws *workspacev1.Workspace) bool {
+		mgr, ok := ws.Labels[k8s.WorkspaceManagedByLabel]
+		if !ok {
+			return true
+		}
+
+		return mgr == constants.ManagedBy
+	}
+
 	// we need several reconciliation loops during a workspace creation until it reaches a stable state.
 	// this introduces the side effect of multiple notifications to the subscribers with partial information.
 	// the filterByUpdate predicate acts as a filter to avoid this
 	filterByUpdate := predicate.Funcs{
 		CreateFunc: func(ce event.CreateEvent) bool {
-			return true
+			ws := ce.Object.(*workspacev1.Workspace)
+			return filterByManagedBy(ws)
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			old := e.ObjectOld.(*workspacev1.Workspace)
 			new := e.ObjectNew.(*workspacev1.Workspace)
 
+			mgr, ok := new.Labels[k8s.WorkspaceManagedByLabel]
+			if !ok {
+				return true
+			}
+
+			if mgr != constants.ManagedBy {
+				return false
+			}
+
```

Comment:
```suggestion
			if !filterByManagedBy(ws) {
				return false
			}

```
that way we still run the filters below as well